From 06fc4e4d4c60f2d2aa40e81fd5eeb0dad17c63ae Mon Sep 17 00:00:00 2001
From: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Date: Mon, 8 Jun 2026 07:49:14 +0000
Subject: [PATCH] HID: i2c-hid-of: fix NULL-deref on failed power up

jira VULN-36624
cve CVE-2024-26717
commit-author Johan Hovold <johan+linaro@kernel.org>
commit 00aab7dcb2267f2aef59447602f34501efe1a07f

A while back the I2C HID implementation was split in an ACPI and OF
part, but the new OF driver never initialises the client pointer which
is dereferenced on power-up failures.

Fixes: b33752c30023 ("HID: i2c-hid: Reorganize so ACPI and OF are separate modules")
	Cc: stable@vger.kernel.org      # 5.12
	Cc: Douglas Anderson <dianders@chromium.org>
	Signed-off-by: Johan Hovold <johan+linaro@kernel.org>
	Reviewed-by: Douglas Anderson <dianders@chromium.org>
	Signed-off-by: Jiri Kosina <jkosina@suse.com>
(cherry picked from commit 00aab7dcb2267f2aef59447602f34501efe1a07f)
	Signed-off-by: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
---
 drivers/hid/i2c-hid/i2c-hid-of.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/hid/i2c-hid/i2c-hid-of.c b/drivers/hid/i2c-hid/i2c-hid-of.c
index 855f53092f4e..4dbbce6fbf34 100644
--- a/drivers/hid/i2c-hid/i2c-hid-of.c
+++ b/drivers/hid/i2c-hid/i2c-hid-of.c
@@ -87,6 +87,7 @@ static int i2c_hid_of_probe(struct i2c_client *client)
 	if (!ihid_of)
 		return -ENOMEM;
 
+	ihid_of->client = client;
 	ihid_of->ops.power_up = i2c_hid_of_power_up;
 	ihid_of->ops.power_down = i2c_hid_of_power_down;
 
-- 
2.47.3

